subscription-service/src/main/kotlin/com/egm/stellio/subscription/service/SubscriptionService.kt

```diff
@@ -269,25 +281,30 @@ class SubscriptionService(
     suspend fun getContextsForSubscription(id: URI): Either<APIException, List<String>> {
         val selectStatement =
             """
-            SELECT contexts
+            SELECT contexts, jsonld_context
             FROM subscription 
             WHERE id = :id
             """.trimIndent()
 
         return databaseClient.sql(selectStatement)
             .bind("id", id)
             .oneToResult {
-                toList(it["contexts"]!!)
+                it["jsonld_context"]?.let { listOf(it as String) } ?: toList(it["contexts"]!!)
             }
     }
 
-    fun getContextsLink(subscription: Subscription): String =
-        if (subscription.contexts.size > 1) {
-            val linkToRetrieveContexts = subscriptionProperties.stellioUrl +
-                "/ngsi-ld/v1/subscriptions/${subscription.id}/context"
-            buildContextLinkHeader(linkToRetrieveContexts)
-        } else
-            buildContextLinkHeader(subscription.contexts[0])
+    fun getContextsLink(subscription: Subscription): String {
+        val contextLink = when {
+            subscription.contexts.size > 1 && subscription.jsonldContext == null -> {
+                val linkToRetrieveContexts = subscriptionProperties.stellioUrl +
+                    "/ngsi-ld/v1/subscriptions/${subscription.id}/context"
+                linkToRetrieveContexts
+            }
+            subscription.jsonldContext != null -> subscription.jsonldContext.toString()
```

Comment:
If you do this check first, you will not need the second condition in the previous case (better readability)